cloudformation/intrinsics.go

```diff
@@ -156,16 +176,28 @@ func SubVars(value interface{}, variables map[string]interface{}) string {
 	return encode(fmt.Sprintf(`{ "Fn::Sub" : [ %q, { %s } ] }`, value, strings.Join(pairs, ",")))
 }
 
+func SubVarsPtr(value interface{}, variables map[string]interface{}) *string {
+	return String(SubVars(value, variables))
+}
+
 // (str, str) -> str
 
 // GetAtt returns the value of an attribute from a resource in the template.
 func GetAtt(logicalName string, attribute string) string {
 	return encode(fmt.Sprintf(`{ "Fn::GetAtt" : [ %q, %q ] }`, logicalName, attribute))
 }
 
+func GetAttPtr(logicalName string, attribute string) *string {
+	return String(GetAtt(logicalName, attribute))
+}
+
 // Split splits a string into a list of string values so that you can select an element from the resulting string list, use the Fn::Split intrinsic function. Specify the location of splits with a delimiter, such as , (a comma). After you split a string, use the Fn::Select function to pick a specific element.
 func Split(delimiter, source interface{}) string {
-	return encode(fmt.Sprintf(`{ "Fn::Split" : [ %q, %q ] }`, delimiter, source))
+	return fmt.Sprintf(`{ "Fn::Split" : [ %q, %q ] }`, delimiter, source)
+}
+
+func SplitPtr(delimiter, source interface{}) *[]string {
```

Comment:
I used here *[]string because we are using Split when we would like to get an array of objects